File: LL.java
import java.util.Random;
public class Main {

    static class ll {
        Node head;
        int size;

        ll() {
            this.size = 0;
        }

        class Node {
            int data;
            Node next;

            Node(int data) {
                this.data = data;
                this.next = null;
                size++;
            }
        }

        public void deleteFirst() {
            if (head == null) {
                System.out.println("The list is empty");
                return;
            }
            size--;
            head = head.next;
        }

        public void addLast(int data) {
            Node newNode = new Node(data);
            if (head == null) {
                head = newNode;
                return;
            }
            Node currNode = head;
            while (currNode.next != null) {
                currNode = currNode.next;
            }
            currNode.next = newNode;
        }

        public int getSize() {
            return size;
        }

        public int getElement(int indx) {
            int i = 1;
            Node curr = head;
            while (i < indx) {
                curr = curr.next;
                i++;
            }
            return curr.data;
        }

        public void deleteLast() {
            if (head == null) {
                System.out.println("The list is empty");
                return;
            }
            size--;
            if (head.next == null) {
                head = null;
                return;
            }
            Node secondLast = head;
            Node lastNode = head.next;
            while (lastNode.next != null) {
                lastNode = lastNode.next;
                secondLast = secondLast.next;

            }
            secondLast.next = null;
        }

        public void deleteatindex(int indx) {
            if (indx == 1) {
                head = head.next;
                return;
            }
            Node curr = head;
            int i = 1;
            while (i < indx - 1) {
                curr = curr.next;
                i++;
            }
            curr.next = curr.next.next;
            size--;
        }




    public void print() {
        Node currNode = head;
        while (currNode != null) {
            System.out.print(currNode.data + "-->");
            currNode = currNode.next;
        }
        System.out.println("null");
    }

}
    public static void main(String[] args) {
        ll list=new ll();
        for(int i=0;i<10;i++){
            Random random = new Random();
            int randomNumber = random.nextInt(50) + 1;
            list.addLast(randomNumber);
        }
        list.print();
        for(int i=0;i<list.getSize();i++){
            if(list.getElement(i)>25) {
                list.deleteatindex(i);
            }
        }
        list.print();

    }
}

File: LL.cpp
#include<iostream>
using namespace std;
class Node{
    public:
    int data;
    Node* next;
    Node(int data){
        this->data=data;
        this->next=NULL;
    }
    ~Node(){
		int value = this->data;
		if(this->next != NULL){
			delete next;
			this->next = NULL;
		}
		cout<<"Memory is free"<<endl;
	}

};

void AddFirst(Node* &head,int d){
    Node* newNode=new Node(d);
    newNode->next=head;
    head=newNode;
}


void AddLast(Node* &head,int d)
{
    Node*newNode=new Node(d);
    Node*temp=head;
    while(temp->next!=NULL){
        temp=temp->next;
    }
    temp->next=newNode;
}
void deleteAtindex(Node* &head,int indx)
{
    if(indx==1){
    Node*newhead=head->next;
    head=newhead;
    return;
             }
    int i=1;
    Node* currNode=head;

    while(i<indx-1&&currNode!=NULL){
        currNode=currNode->next;
          i++;
         }
currNode->next=currNode->next->next;
}

void insertatposition(Node* &head,int indx,int data){
    Node* newNode=new Node(data);
    
    if(indx==1){
        Node* temp=head->next;
        head=newNode;
        newNode->next=temp;
         return ;
    }
    int i=1;
    Node*temp=head;
    while(i<indx-1&&temp!=NULL){
        temp=temp->next;
        i++;
    }
    Node* nextNode=temp->next;
    temp->next=newNode;
    newNode->next=nextNode;

}

void print(Node* &head){
    Node* currNode=head;
    while(currNode!=NULL){
        cout<<currNode->data<<"-->";
        currNode=currNode->next;
    }
    cout<<"NULL\n";
}
int main(){
    Node*head=new Node(10);
    AddFirst(head,20);
    AddFirst(head,30);
    print(head);
    cout<<"Adding 0 at last\n";
    AddLast(head,0);
    print(head);
    cout<<"removing 1st node\n";
    deleteAtindex(head,1);
    print(head);
    cout<<"adding node at 2nd position\n";
    insertatposition(head,2,80);
    print(head);
    
    return 0;
    }
